app/grandchallenge/products/templates/products/partials/product_image_carousel.html

```diff
@@ -1,31 +1,33 @@
 {% if product.images.all %}
-  <div id="product_img_carousel" class="carousel slide" data-ride="carousel" data-interval="false">
-    <ol class="carousel-indicators">
-      {% for image in product.images.all %}
-        <li data-target="#product_img_carousel" data-slide-to={{ forloop.counter0 }} {% if forloop.counter0 == 0%}class="active"{% endif %}></li>
-      {% endfor %}
-    </ol>
-    <div class="carousel-inner">
-      {% for image in product.images.all %}
-        <div class="carousel-item {% if forloop.counter0 == 0 %}active{% endif %}">
-          <img class="d-block w-100" src={{ image.img.url }}>
-        </div>
-      {% endfor %}
+  {% if product.images.all.count == 1 %}
+    <img src={{product.images.first.img.url}} style="width:100%;">
+  {% else %}
+    <div id="product_img_carousel" class="carousel slide" data-ride="carousel" data-interval="false">
+      <ol class="carousel-indicators">
+        {% for image in product.images.all %}
+          <li data-target="#product_img_carousel" data-slide-to={{ forloop.counter0 }} {% if forloop.counter0 == 0%}class="active"{% endif %}></li>
+        {% endfor %}
+      </ol>
+      <div class="carousel-inner">
+        {% for image in product.images.all %}
```

Comment:
This is a bit nitpicky, but you could also create variable for `product.images.all`, as you currently evaluate the queryset twice. You could either do this in the views `get_context_data`, or by using `{% with images=product.images.all %}` in the template.

This is not something that I think has to be changed, but it is something to be mindful of when writing django code.